### PR TITLE
chore: added swagger types for BOM operations

### PR DESCRIPTION
**Pull request by omerlh**
### Description

I was working on generating a typescript client using Swagger (happy to contribute the client also!) and I noticed some APIs does not have well defined responses.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

I am not sure how I should test it, so happy to get some feedback on the change!

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Can I suggest to submit a PR with only the changes you propose? Currently 90% seems to be only linting?

**Comment by omerlh** [comment]
Yes, sorry - fixed. VS code did things I didn't noticed...

**Comment by nscuro** [comment]
You'll find lots of these inconsistencies, which is why I can't really recommend generating clients from the swagger doc. See #840.

**Comment by omerlh** [comment]
I don't mind keep fixing it as I'm going, I have a client that works pretty well and generated from Swagger. If this is acceptable by you, OFC...

**Review by nscuro — CHANGES_REQUESTED** [review]
I guess I'm fine with small incremental improvements until we have a better OpenAPI doc in place. 

A few minor adjustments and clarifications.

File: src/main/java/org/dependencytrack/model/PortfolioMetrics.java [hunk 1]
```diff
@@ -51,18 +53,22 @@ public class PortfolioMetrics implements Serializable {
 
     @Persistent
     @Column(name = "CRITICAL")
+    @NotNull
```
[the remark this answers is not in the document]
Comment (omerlh) [reply src/main/java/org/dependencytrack/model/PortfolioMetrics.java:56]: Yes, I wasn't sure which one is better to use, I can change it.

File: src/main/java/org/dependencytrack/resources/v1/vo/BomUploadResponse.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.dependencytrack.resources.v1.vo;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import io.swagger.annotations.ApiModelProperty;
+
+public class BomUploadResponse implements Serializable {
+
+    private static final long serialVersionUID = -7592438786581683855L;
```
[the remark this answers is not in the document]
Comment (omerlh) [reply src/main/java/org/dependencytrack/resources/v1/vo/BomUploadResponse.java:10]: Yes, I copied it from another class and just changed some numbers. Changing more numbers now.

**Comment by omerlh** [comment]
I think the build failure are not related to my changes

**Comment by nscuro** [comment]
@omerlh Sorry about that, #2241 should fix the build failure. If you rebase with `master` it should work again.

**Comment by omerlh** [comment]
Ok, rebased

**Review by nscuro — APPROVED** [review]
Thanks @omerlh! 🚀

**Pull request merged**